src/Analyzers/Analyzers/src/UseParameterAttributeCorrectlyAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+            id: "ASP0002",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+            title: "A property that has 'ParameterAttribute' should be an auto-property",
+            messageFormat: "The property '{0}' should be auto-property",
+            category: "",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.GetTypeByMetadataName(WellKnownTypeNames.MicrosoftAspNetCoreComponentsParameterAttribute) is INamedTypeSymbol parameterAttributeSymbol)
+                {
+                    context.RegisterSymbolAction(async context =>
+                    {
+                        var symbol = (IPropertySymbol)context.Symbol;
+                        if (symbol.HasAttribute(parameterAttributeSymbol) && !symbol.IsAutoProperty() &&
+                            !await IsSameSemanticAsAutoPropertyAsync(symbol, context.CancellationToken).ConfigureAwait(false))
+                        {
+                            context.ReportDiagnostic(Diagnostic.Create(s_rule, symbol.Locations.FirstOrDefault(), symbol.Name));
+                        }
+                    }, SymbolKind.Property);
+                }
+            });
+        }
+
+        private async Task<bool> IsSameSemanticAsAutoPropertyAsync(IPropertySymbol symbol, CancellationToken cancellationToken)
+        {
+            // This is not the preferred way to do things. There is a current work to support C# and VB with separate projects.
+            // When that's done, this should be made abstract and have different C# and VB implementations.
+            if (symbol.Language == LanguageNames.CSharp && symbol.DeclaringSyntaxReferences.Length == 1 &&
+                await symbol.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false) is PropertyDeclarationSyntax syntax &&
+                syntax.AccessorList.Accessors.Count == 2)
+            {
+                var getterAccessor = syntax.AccessorList.Accessors[0];
+                var setterAccessor = syntax.AccessorList.Accessors[1];
+                if (getterAccessor.IsKind(SyntaxKind.SetAccessorDeclaration))
+                {
+                    // Swap if necessary.
+                    (getterAccessor, setterAccessor) = (setterAccessor, getterAccessor);
+                }
+
+                if (!getterAccessor.IsKind(SyntaxKind.GetAccessorDeclaration) || !setterAccessor.IsKind(SyntaxKind.SetAccessorDeclaration))
```

Comment:
I didn't read the rest of this method, but is there a way to detect this by inspecting if the getters / setters are compiled generated? 

@sharwell in case you have suggestions. We're basically trying to determine that a property is an auto-property. `public string Prop1 { get; set; }`

Reply:
@pranavkm This method isn't for detecting auto property, but full-properties with the same semantics. See the comments in the linked issue.